lib/manager/components/version/FeedVersionViewer.js

```diff
@@ -51,13 +55,32 @@ export type Props = {
 export default class FeedVersionViewer extends Component<Props> {
   messages = getComponentMessages('FeedVersionViewer')
 
+  componentWillMount () {
+    const {fetchValidationIssueCount, validateGtfsPlusFeed, version} = this.props
+    // Fetch validation issue count when the component mounts. The validation
+    // issue counts are needed to determine whether any blocking/major issues
+    // exist with the version.
+    if (version) {
+      fetchValidationIssueCount(version)
+      validateGtfsPlusFeed(version.id)
+    }
+  }
+
+  componentWillReceiveProps (nextProps: Props) {
+    const {fetchValidationIssueCount} = this.props
+    if (nextProps.version && !isErrorCountFetched(nextProps.version.validationResult)) {
+      fetchValidationIssueCount(nextProps.version)
+    }
+  }
+
   render () {
     const {
       feedVersionIndex,
       listView,
       newNotePosted,
       notesRequested,
       version,
+      versions,
```

Comment:
Any of the components in this `render` function that get passed `{...this.props}` really ought to be connected components.